ex1/functions.py

```diff
@@ -90,26 +135,98 @@ def f():
 # Define a function named args_to_dict that takes three arguments and returns a
 # dictionary with the position of the argument as the key (starting at 0) and
 # the argument as the value.
-
+def args_to_dict(arg1, arg2, arg3):
+    loc = locals()    
+    kwargs = dict()
+    i = 0    
+    for arg in loc:
+        kwargs[i]=loc.get(arg)
+        i+=1
+    return kwargs
+    
 # BONUS: Write a function named args_to_dict_general that does the same for any
 # number of arguments
+def args_to_dict_general(*args):
+    loc = locals()    
+    kwargs = dict()
+    i = 0    
+    for args in loc:
```

Comment:
There is really no need to loop over the variables in the local namespace. You can just loop over `args` and it should work.